mmeval/metrics/niqe.py

```diff
@@ -0,0 +1,462 @@
+            os.path.join(os.path.dirname(__file__), 'niqe_pris_params.npz'))
+        mu_pris_param = niqe_pris_params['mu_pris_param']
+        cov_pris_param = niqe_pris_params['cov_pris_param']
+        gaussian_window = niqe_pris_params['gaussian_window']
+
+        if channel_order is None:
+            channel_order = self.channel_order
+
+        for prediction in predictions:
+
+            if len(prediction.shape) <= 3:
+                result = self.compute_niqe(prediction, channel_order,
+                                           mu_pris_param, cov_pris_param,
+                                           gaussian_window)
+            else:
+                result_sum = 0
+                for i in range(prediction.shape[0]):
+                    result_sum += self.compute_niqe(prediction[i],
+                                                    channel_order,
+                                                    mu_pris_param,
+                                                    cov_pris_param,
+                                                    gaussian_window)
+                result = result_sum / prediction.shape[0]
+
+            self._results.append(result)
+
+    def compute_metric(self, results: List[np.float64]) -> Dict[str, float]:
+        """Compute the NIQE metric.
+
+        This method would be invoked in ``BaseMetric.compute`` after
+        distributed synchronization.
+
+        Args:
+            results (List[np.float64]): A list that consisting the NIQE score.
+                This list has already been synced across all ranks.
+
+        Returns:
+            Dict[str, float]: The computed NIQE metric.
+        """
+
+        return {'niqe': float(np.array(results).mean())}
+
+    def compute_niqe(self,
+                     prediction: np.ndarray,
+                     channel_order: str,
+                     mu_pris_param: np.ndarray,
+                     cov_pris_param: np.ndarray,
+                     gaussian_window: np.ndarray,
+                     block_size_h: int = 96,
+                     block_size_w: int = 96) -> np.float64:
+        """Calculate NIQE (Natural Image Quality Evaluator) metric. We use the
+        official params estimated from the pristine dataset. We use the
+        recommended block size (96, 96) without overlaps.
+
+        Note that we do not include block overlap height and width, since they
+        are always 0 in the official implementation.
+
+        For good performance, it is advisable by the official implementation to
+        divide the distorted image in to the same size patched as used for the
+        construction of multivariate Gaussian model.
+
+        Args:
+            prediction (np.ndarray): Input image whose quality to be computed.
+                The image must be a gray or Y (of YCbCr) image with shape
+                (h, w). Range [0, 255] with float type.
+            channel_order (str): The channel order of image.
```

Comment:
I suspect `prediction` should not be a gray or Y image here, since there is a `channel_order` argument.